src/Lecture1.hs

```diff
@@ -98,7 +111,14 @@ start position can be considered as zero (e.g. substring from the
 first character) and negative end position should result in an empty
 string.
 -}
-subString start end str = error "TODO"
+subString :: Int -> Int -> String -> String
+subString start end str = 
+    let 
+        begin 
+            | start < 0  = 0
+            | otherwise  = start
```

Comment:
It's a nice usage of guards inside `let` 👌🏻 
I haven't explained this syntax but it works as you noticed 😅 

You can write this slightly shorter though by using standard function `max`:

```suggestion
        begin = max 0 start
```